modules/l4tls/alpn_matcher.go

```diff
@@ -37,10 +37,12 @@ func (MatchALPN) CaddyModule() caddy.ModuleInfo {
 }
 
 func (m MatchALPN) Match(hello *tls.ClientHelloInfo) bool {
+	repl := hello.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
 	clientProtocols := hello.SupportedProtos
-	for _, alpn := range m {
+	for _, s := range m {
+		rs := repl.ReplaceAll(s, "")
```

Comment:
```suggestion
		s = repl.ReplaceAll(s, "")
```

Since I don't think we need `s` before the replacement.

Do you prefer this variable name to `alpn` as it originally was?

Reply:
Thanks, I've fixed this issue here and in similar places of the code.